pkgs/os-specific/linux/openrazer/daemon.nix

```diff
@@ -0,0 +1,66 @@
+{ buildPythonApplication
+, daemonize
+, dbus-python
+, fetchFromGitHub
+, gobjectIntrospection
+, gtk3
+, makeWrapper
+, pygobject3
+, pyudev
+, setproctitle
+, stdenv
+}:
+
+let
+  openrazerSrc = import ./src.nix;
+in
+buildPythonApplication rec {
+  inherit (openrazerSrc) version;
+  pname = "openrazer_daemon";
+
+  outputs = [ "out" "man" ];
+
+  src = fetchFromGitHub openrazerSrc.github;
+  sourceRoot = "source/daemon";
+  patches = [
+    # https://github.com/openrazer/openrazer/pull/681
+    ./openrazer_python_daemon_script.patch
```

Comment:
Instead of storing the patches in nixpkgs you might be able to refer to them using:
```nix
(fetchpatch {
  url = https://github.com/openrazer/openrazer/commit/feefb86497af6ff7db827e682ae560c12d0aee2f.patch;
  sha256 = "...";
})
```

Reply:
The patches from the pull requests are relative to the entire openrazer repository. The patches stored in nixkpgs are relative to the sourceRoot `"source/daemon"`. We could of course take the patches from github and rewrite them to be relative to `"daemon"`.

Reply:
You can use [the `stripLen` option of `fetchpatch`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/fetchpatch/default.nix#L8) to strip of the "daemon/" prefix like:

```nix
(fetchpatch {
  url = https://github.com/openrazer/openrazer/commit/feefb86497af6ff7db827e682ae560c12d0aee2f.patch;
  sha256 = "...";
  stripLen = 1;
})
```

Reply:
Done